app/Http/Controllers/Front/CartController.php

```diff
@@ -183,6 +192,29 @@ public function showCart()
         }
     }
 
+    private function checkUnpaidInvoices($item)
+    {
+        $userId = \Auth::user()->id;
```

Comment:
Too much code just do this 

 $userId = auth()->id();
    $itemName = $item->name;
    $itemQuantity = $item->quantity;

    return Invoice::where('user_id', $userId)
        ->where('is_renewed', 0)
        ->where('status', 'pending')
        ->whereHas('invoiceItem', function ($query) use ($itemName, $itemQuantity) {
            $query->where('product_name', $itemName)->where('quantity', $itemQuantity);
        })
        ->firstOr(function () use ($item) {
            Cart::clear($item->id);
        });